### ISSUE TITLE
[data grid] Duplicating columns when grouping.

### ISSUE THREAD

**Issue by RomanMain** [mui/mui-x#12223]
### The problem in depth

When grouping a specific column, we have automatically generated new columns that are added to the filtering list and the “manage columns” section. Their names are completely duplicated. Is there some option to prevent these duplicate columns from appearing in these lists?

[https://stackblitz.com/run?file=Demo.js](https://stackblitz.com/run?file=Demo.js)

![Screenshot_1](https://github.com/mui/mui-x/assets/43311617/9b06f3e2-81b9-4261-83d4-7b79fb93dad7)
![Screenshot_2](https://github.com/mui/mui-x/assets/43311617/b4ae688b-e52c-4c02-9d04-9194da2de08b)

### Your environment

<details>
  <summary>`npx @mui/envinfo`</summary>

```
  Don't forget to mention which browser you used.
  Output from `npx @mui/envinfo` goes here.
```
</details>

**Search keywords**: grouping, duplicate columns
**Order ID**: 82865

**Comment by michelengelen** [comment]
Hey @RomanMain and thanks for raising this issue.
It is indeed not entirely intuitive and I guess even a bug, right @MBilalShafi ?

For a workaround (and maybe for additional filtering if needed) you can use the `slotProps.filterPanel.filterFormProps.filterColumns` method:

```tsx
const filterColumns = ({ columns }: FilterColumnsArgs) => {
  return columns.filter((column) => column.type !== 'custom').map((column) => column.field);
};

...

<DataGridPremium
  {...data}
  apiRef={apiRef}
  initialState={initialState}
  onRowSelectionModelChange={handleRowSelectionModelChange}
  rowSelectionModel={rowSelectionModel}
  checkboxSelection
  hideFooter
  slotProps={{
    filterPanel: {
      filterFormProps: {
        filterColumns,
      },
    },
  }}
/>
```

In this specific case we can just filter by `column.type !== 'custom'` since those will filter all auto generated columns by default.

I will still put this on our board to see if the current behavior is intended.
If you need additional assistance please feel free to reach out! Thanks again! 🙇🏼

**Comment by RomanMain** [comment]
Thanks for the quick response @michelengelen! This works for the filter list, but how can I remove this in the "Manage columns"  panel as well? I can’t see a slotProps similar to this in the documentation.

**Comment by michelengelen** [comment]
> Thanks for the quick response @michelengelen! This works for the filter list, but how can I remove this in the "Manage columns" panel as well? I can’t see a slotProps similar to this in the documentation.

No problem ... I did forget to include that.
There actually is a recipe that shows how to do this in the docs: [Customize the list of columns in panel](https://mui.com/x/react-data-grid/column-visibility/#customize-the-list-of-columns-in-panel)

TLDR;
```tsx
const getTogglableColumns = (columns: GridColDef[]) => {
  return columns.filter((column) => column.type !== 'custom').map((column) => column.field);
};

...

<DataGridPremium
  {...data}
  apiRef={apiRef}
  initialState={initialState}
  onRowSelectionModelChange={handleRowSelectionModelChange}
  rowSelectionModel={rowSelectionModel}
  checkboxSelection
  hideFooter
  slotProps={{
    filterPanel: {
      filterFormProps: {
        filterColumns,
      },
    },
    columnsManagement: {
      getTogglableColumns,
    },
  }}
/>
```

**Comment by MBilalShafi** [comment]
Hey @RomanMain,

As @michelengelen mentioned, the [Customize the list of columns in panel](https://mui.com/x/react-data-grid/column-visibility/#customize-the-list-of-columns-in-panel) seems exactly what you are looking for. The demo depicts how to hide the `status` column, and the grouped column from the columns panel (v6) and columns management panel (v7).
